### PR TITLE
Improve list formatter tests

### PR DESCRIPTION
**Pull request by sffc**
See #5107

### CODE REVIEW

File: components/list/src/lazy_automaton.rs [hunk 1]
```diff
@@ -47,7 +47,7 @@ impl<T: AsRef<[u8]>> LazyAutomaton for DFA<T> {
     }
 }
 
-#[cfg(test)]
+#[cfg(all(test, feature = "datagen"))]
```
[the remark this answers is not in the document]
Comment (sffc) [reply components/list/src/lazy_automaton.rs:50]: We don't feature-gate docs tests but it's fine for unit tests, especially now that there's the lint that prevents features from being spelled wrong

File: components/list/src/list_formatter.rs [hunk 2]
```diff
@@ -326,7 +328,7 @@ mod tests {
     fn test_into_iterator() {
         let formatter = formatter(ListLength::Wide);
 
-        let mut vecdeque = std::collections::vec_deque::VecDeque::<u8>::new();
+        let mut vecdeque = alloc::collections::vec_deque::VecDeque::<u8>::new();
```
[the remark this answers is not in the document]
Comment (sffc) [reply components/list/src/list_formatter.rs:331]: It's not available if the tests are not run with all features

I think our conclusion was that tests without all features could be supported on a best effort basis, which is what I did here

File: components/list/src/lazy_automaton.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (robertbastian) [reply components/list/src/lazy_automaton.rs:50]: This is an unnecessary change that is unrelated to what this PR says it's doing.

[the remark this answers is not in the document]
Comment (sffc) [reply components/list/src/lazy_automaton.rs:50]: I changed the title of the PR

**Review by robertbastian — CHANGES_REQUESTED** [review]
These tests don't add anything, we already have a test that uses fallback with Spanish, and that fails with the setup in https://github.com/unicode-org/icu4x/issues/5107.

**Pull request closed without merging**